R/compengine.R

```diff
@@ -0,0 +1,815 @@
+
+#' CompEngine feature set
+#'
+#' Calculate the features that have been used in CompEngine database, using method introduced in package
+#' \code{kctsa}. 
+#' 
+#' The features involved can be grouped as \code{autocorrelation}, 
+#' \code{prediction}, \code{stationarity}, \code{distribution}, and \code{scaling}. 
+#'
+#' @param y the input time series
+#' @return a vector with CompEngine features
+#' @seealso \code{\link{autocorr_features}}
+#' @seealso \code{\link{pred_features}}
+#' @seealso \code{\link{station_features}}
+#' @seealso \code{\link{dist_features}}
+#' @seealso \code{\link{scal_features}}
+#' @references B.D. Fulcher and N.S. Jones. hctsa: A computational framework for automated time-series phenotyping using massive feature extraction. Cell Systems 5, 527 (2017).
+#' @references B.D. Fulcher, M.A. Little, N.S. Jones Highly comparative time-series analysis: the empirical structure of time series and their methods. J. Roy. Soc. Interface 10, 83 (2013).
+#' @author Yangzhuoran Yang
+#' @export
+compengine <- function(x){
+  c(autocorr_features(x), pred_features(x), station_features(x), dist_features(x), scal_features(x))
+}
+
+
+#' The autocorrelation feature set from software package \code{hctsa}
+#'
+#' Calculate the features that grouped as autocorrelation set, 
+#' which have been used in CompEngine database, using method introduced in package \code{kctsa}. 
+#' 
+#' Features in this set are \code{Embed2_incircle_1}, 
+#' \code{Embed2_incircle_2}, 
+#' \code{AC_9}, 
+#' \code{FirstMin_ac}, 
+#' \code{trev_num}, 
+#' \code{MotifTwo_entro3}, 
+#' and \code{Walker_propcross}.
+#'
+#' @param y the input time series
+#' @return a vector with autocorrelation features
+#' @seealso \code{\link{Embed2_incircle}}
+#' @seealso \code{\link{AC_9}}
+#' @seealso \code{\link{FirstMin_ac}}
+#' @seealso \code{\link{trev_num}}
+#' @seealso \code{\link{MotifTwo_entro3}}
+#' @seealso \code{\link{Walker_propcross}}
+#' @references B.D. Fulcher and N.S. Jones. hctsa: A computational framework for automated time-series phenotyping using massive feature extraction. Cell Systems 5, 527 (2017).
+#' @references B.D. Fulcher, M.A. Little, N.S. Jones Highly comparative time-series analysis: the empirical structure of time series and their methods. J. Roy. Soc. Interface 10, 83 (2013).
+#' @author Yangzhuoran Yang
+#' @export
+autocorr_features <- function(x){
```

Comment:
This is inefficient. Many of these functions call the acf() function on the same data.

Reply:
I add a line next to it calculating the acf once
 `acfv <- stats::acf(x, length(x)-1, plot = FALSE)`
and then refer to the stored value using an argument whenever an acf call is required
`embed2_incircle_2 = embed2_incircle(x,2, acfv = acfv),`
so it won't repeatedly call the acf() function in each feature function.
Is this good enough? If not, could you give me some ideas?

Reply:
Yes, that will work.